docs/authentication/stytch/permit-integration.mdx

```diff
@@ -0,0 +1,406 @@
+import { useRouter } from "next/router";
+
+export default function ProfilePage() {
+  const { user, isInitialized } = useStytchUser();
+  const router = useRouter();
+
+  useEffect(() => {
+    if (isInitialized && !user) {
+      router.replace("/");
+    }
+  }, [user, isInitialized, router]);
+
+  return <div>Profile Page Content</div>;
+}
+```
+
+### Step 6: Grant Access to Protected Server-side Routes
+
+The below code will allow you to work with Stytch's backend Node SDK, which will retrieve the Stytch `session_token` or `session_jwt`
+from the request cookies, and then call the `sessions.authenticate` or `sessions.authenticateJwt` methods:
+
+```javascript
+import stytch from "stytch";
+
+let client;
+
+const loadStytch = () => {
+  if (!client) {
+    client = new stytch.Client({
+      project_id: process.env.STYTCH_PROJECT_ID || "",
+      secret: process.env.STYTCH_SECRET || "",
+      env: process.env.STYTCH_PROJECT_ENV === "live" ? stytch.envs.live : stytch.envs.test,
+    });
+  }
+  return client;
+};
+
+export async function getServerSideProps({ req }) {
+  const redirectRes = {
+    redirect: {
+      destination: "/",
+      permanent: false,
+    },
+  };
+
+  const sessionJWT = req.cookies["stytch_session_jwt"];
+
+  if (!sessionJWT) {
+    return redirectRes;
+  }
+
+  const stytchClient = loadStytch();
+
+  try {
+    await stytchClient.sessions.authenticateJwt(sessionJWT);
+    return { props: {} };
+  } catch (e) {
+    return redirectRes;
+  }
+}
+```
+
+As you saw in the above step, we are fetching a `sessionJWT` directly from the cookies. As we are also able to authenticate the session
+with the `.authenticateJwt()` function, we therefore know the `user id` will also be available to us, so we can proceed to sync the user
+into Permit. This is part of the **handoff point** implementation.
+
+:::tip WHY DO WE SYNC USERS?
+We need to sync user identities into Permit to understand each user's assigned role. This allows us to immediately identify their
+role upon successful authentication, enabling us to apply the appropriate policies and rules. As a result, we can clearly define
+what each user is permitted or restricted from doing.
+:::
+
+Before we move to utilize the syncUser function, we need to do a basic Permit setup in the UI.
+
+## Permit Pre-sync setup guide
+
+For the initial setup guide with Permit, please follow the [Quickstart guide](/quickstart). It will give you an outline how to
+create a basic `workspace` with permit, create roles, resources, and assign permissions to the roles.
+
+Below is an example of the policy we will be working with. The only thing we are missing is syncing a user into Permit, and assigning
+them the appropriate role.
+
+![Basic Policy](/ui-videos/authentication/stytch/account-owner-role.png)
+
+### Importing and initializing the Permit object
+
+As you work with your app's implementation, any critical, data sensitive actions are performed in the backend. Inside of your backend
```

Comment:
Can you please propose where this code should sit? I think `your backend file` is missing context